astropy/coordinates/sky_coordinate.py

```diff
@@ -483,6 +485,60 @@ def transform_to(self, frame, merge_attributes=True):
             frame_kwargs.pop(attr)
         return self.__class__(new_coord, **frame_kwargs)
 
+    @u.quantity_input(dt=(u.year, None))
```

Comment:
Wouldn't do this: there's nothing against passing in a `TimeDelta`. I'd just duck type: things will fail with an appropriate unit error if something wrong is passed in.